WcaOnRails/spec/lib/results_validators/competitor_limit_validator_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RV = ResultsValidators
+CLV = RV::CompetitorLimitValidator
+
+RSpec.describe CLV do
+  context "on InboxResult and Result" do
+    let!(:competition1) { FactoryBot.create(:competition, :past, :with_competitor_limit, event_ids: ["333oh"], competitor_limit: 12) }
+    let!(:competition2) { FactoryBot.create(:competition, :past, :with_competitor_limit, event_ids: ["222"], competitor_limit: 10) }
+
+    # The idea behind this variable is the following: the validator can be applied
+    # on either a particular model for given competition ids, or on a set of results.
+    # We simply want to check it has the expected behavior on all the possible cases.
+    let(:validator_args) {
+      [InboxResult, Result].flat_map { |model|
+        [
+          { competition_ids: [competition1.id, competition2.id], model: model },
+          { results: model.sorted_for_competitions([competition1.id, competition2.id]), model: model },
+        ]
+      }
+    }
+
+    context "for competitions having a competitor limit" do
+      before(:example) do
+        [Result, InboxResult].each do |model|
+          result_kind = model.model_name.singular.to_sym
+          10.times {
+            FactoryBot.create(result_kind, competition: competition1, eventId: "333oh")
+            FactoryBot.create(result_kind, competition: competition2, eventId: "222")
```

Comment:
Perhaps we could use `FactoryBot.create_list(result_kind, 10, ...)` here?

Reply:
Yes absolutely!